### PR TITLE
feat: Auth Page

### LINKED ISSUE #2: Auth
**[Figma Link](https://www.figma.com/file/yO3edbsblDYwj42L8yYLrI/CodeCharacter-2023?node-id=80%3A120&t=GTLykSBvdJnUQFqI-0)**
## Add auth pages : 
   1. Sign-up:
        - Accepting details of user ( username,full name,email-id,password,college,country and avatar) along with reCAPTCHA v2.
   2. Login: 
        - Jwt Login ( Accepts email-id and password from user)
        - Google oauth sign-in
        - GitHub oauth sign-in
        - forget-password
  3. Integrate auth with backend.
  4. Display proper ```Toast``` messages for successful responses and errors
  5. Pages should be responsive

### PR DESCRIPTION
**Pull request by Valliammai-SM**

### CODE REVIEW

File: codecharacter-web-2023.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vigneshd332) [inline codecharacter-web-2023.iml]: what is this file? Why is this here?

**Review by vigneshd332 — CHANGES_REQUESTED** [review]

**Comment by vigneshd332** [comment]
image

### Changes

- [x] leave some space between sign in and navbar
- [x] Reduce font size in the buttons and forgot password according to design
- [x] center the oauth buttons vertically
- [x] remove that register in the bottom left corner
- [x] import the fonts you have used in the <code>src/main.module.css</code> file from google fonts or some other cdn
- [x] Forgot password on hover make it pointer
- [x] center that arrow in the login button and make it a little bit bigger
- [x] change the forgot password modal to match the design

image

- [x] Sign Up is in a different position from sign in, please look at the design and see the padding and margin and implement properly
- [x] leave some of margin between buttons
- [x] in next button align the arrow center vertically

image

- [x] Height of the country field
- [x] reduce size of the placeholder font

image

- [x] avatar field should have border like design and transparent background

**Comment by vigneshd332** [comment]
### Changes

- [x] Check the password check in Sign Up, its saying error for valid passwords, also give an informative toast about why the password is invalid. You can give specific toasts like ```Password less than 8 characters``` or ```Password should contain atleast one special character, one number...```
- [x] Update the font in the inputs in sign up
- [x] clicking next in the signup form reloads the page, fix that
- [x] fix mobile responsiveness in the signup page and leave some margin between oauth buttons in login page in mobile
- [x] change the forgot password modal to match the theme

**Comment by Ram-20062003** [comment]
- [x] Give `proper toast messages`( When `password` and `confirm password` does not match  toast `Password and Confirm password doesn't match`) . Change wherever it is necessary
- [x] Add icons in place-holders

**Comment by vigneshd332** [comment]
### Changes

- [x] Remove the alert component and use the toast function
- [x] match font size in sign up to login
- [x] spacing between the oauth buttons and divider is smaller than input fields and divider in sign in page, make them equal
- [x] college field check is not done in register, complete it
- [x] remove unused components like alert and progressbar

**Comment by vigneshd332** [comment]
image
## Changes

- [x] Add some padding left and right to the highlighted description part in mobile.
- [x] Make avatar field size match the other fields.
- [x] make the login button similar to the one in the signup page (with that arrow mark and no border) for both mobile and web.
- [x] Sign Up has some bugs, look into that.

File: src/components/Auth/Auth/Login/IncompleteProfile/incompleteProfile.module.css [hunk 2]
```diff
@@ -0,0 +1,130 @@
+@import '//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css';
```
Comment (vigneshd332) [inline src/components/Auth/Auth/Login/IncompleteProfile/incompleteProfile.module.css]: this is not a link la, remove this if youre not using it or move the import to main.module.css with the correct link if its needed

**Review by vigneshd332 — CHANGES_REQUESTED** [review]

**Review by vigneshd332 — APPROVED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Pull request merged**